### ISSUE TITLE
SIMD accelerated escape routines

### ISSUE THREAD

**Issue by dralley** [tafia/quick-xml#405]
Unlike the unescape routines, the routines for escaping text don't currently utilize any SIMD accelleration.

This should be possible to do via the `jetscii` crate.  `memchr` is currently used by the unescape routines, but while it is supposed to be slightly faster than `jetscii` it is also more limited and can only handle searching for up to 3 different bytes at a time, whereas `jetscii` can handle up to 16.  Since escaping text requires searching for up to 5 characters `<>&" '`, `memchr` is not an option but `jetscii` is.

Benchmark coverage needs to be added first: https://github.com/tafia/quick-xml/issues/404

**Comment by dralley** [comment]
Preview

![image](https://user-images.githubusercontent.com/2751996/176348885-83e8b4b8-79a2-42f2-a821-ddacb4438d86.png)

**Comment by Mingun** [comment]
This is effect from switching to `jetscii`?

**Comment by dralley** [comment]
Yes, and it only requires about 3 lines of change.  I'm going to see if it can be improved any further and whether the occasional regressions can be eliminated.

**Comment by dralley** [comment]
Some reading material, not so much for escape routines specifically but parsing XML (actually HTML) in general

https://lemire.me/blog/2024/06/08/scan-html-faster-with-simd-instructions-chrome-edition/
